### PR TITLE
Core - created display name heuristic

### PR DESCRIPTION
**Pull request by Vojtech-Sassmann**
* There were some known problems with parsing display name. If the name
contained a middle name, it was not parsed properly. Also, sometimes the
display name contained first name and last name in reverse order.
* Sometimes, it is possible to improve this, if the IDP send sn(surname)
and given name of the user.
* Now, if the idp did sent the information, the parser tries to match
these values (also in reverse order). If is succeedes it tries to parse
titles and middle name.

### CODE REVIEW

**Review by balcirakpeter — CHANGES_REQUESTED** [review]
Please add commit message

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java [hunk 1]
```diff
@@ -3323,27 +3353,133 @@ private Candidate createCandidateFromApplicationData(Integer appId) {
...
+	/**
+	 * Generates pattern for parsing titles and middle name from given values.
+	 *
+	 * The pattern is of format: ^(.*){firstName}(.*){lastName}(.*)$
+	 *
+	 * @param firstName first name
+	 * @param lastName last name
+	 * @return pattern for parsing titles and middle name
+	 */
+	private Pattern getNamesPatter(String firstName, String lastName) {
```
Comment (balcirakpeter) [inline perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:3470]: Typo getNamesPatter

**Review by balcirakpeter — APPROVED** [review]
Just the typo, otherwise ok.

**Review by balcirakpeter — APPROVED** [review]

**Review by zlamalp — CHANGES_REQUESTED** [review]

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java [hunk 2]
```diff
@@ -1675,7 +1679,7 @@ public Application approveApplicationInternal(PerunSession sess, int appId) thro
 						// ==> updateNameTitles() in case of change in appForm.
 						updateUserNameTitles(app);
```
[the remark this answers is not in the document]
Comment (Vojtech-Sassmann) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:1680]: We actually want to make some bigger changes to the logic of parsing names. So I think this won't be necessary right now, or we can fix that in some other PR.

File: perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java [hunk 3]
```diff
@@ -158,6 +160,8 @@
 	private static final String NAMESPACE_GROUP_MAIL_FOOTER = AttributesManager.NS_GROUP_ATTR_DEF;
 	static final String URN_GROUP_MAIL_FOOTER = NAMESPACE_GROUP_MAIL_FOOTER + ":" + FRIENDLY_NAME_GROUP_MAIL_FOOTER;
 
+	private static final String FED_SN = "sn";
+	private static final String FED_GIVEN_NAME = "givenName";
```
[the remark this answers is not in the document]
Comment (Vojtech-Sassmann) [reply perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java:164]: I have used the old constants.

**Review by zlamalp — APPROVED** [review]

**Pull request merged**